File: testdata/jennies/rawtypes/disjunctions_of_refs_without_discriminator/JavaRawTypes/disjunctions_of_refs_without_discriminator/TypeB.java
package disjunctions_of_refs_without_discriminator;

import java.util.Objects;

public class TypeB {
    public Long fieldB;
    public TypeB() {
        this.fieldB = 0L;
    }
    public TypeB(Long fieldB) {
        this.fieldB = fieldB;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof TypeB)) return false;
        TypeB o = (TypeB) other;
        if (!Objects.equals(this.fieldB, o.fieldB)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.fieldB);
    }
}

File: testdata/jennies/rawtypes/disjunctions_of_refs_without_discriminator/PHPRawTypes/src/DisjunctionsOfRefsWithoutDiscriminator/TypeB.php
<?php

namespace Grafana\Foundation\DisjunctionsOfRefsWithoutDiscriminator;

class TypeB implements \JsonSerializable
{
    public int $fieldB;

    /**
     * @param int|null $fieldB
     */
    public function __construct(?int $fieldB = null)
    {
        $this->fieldB = $fieldB ?: 0;
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->fieldB !== $other->fieldB) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{fieldB?: int} $inputData */
        $data = $inputData;
        return new self(
            fieldB: $data["fieldB"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->fieldB = $this->fieldB;
        return $data;
    }
}
